### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by tarasavage**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,25 +1,85 @@
...
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: SlideLimitationValidator, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: SlideLimitationValidator
```
Comment (Vasyl-Smutok) [inline app/main.py:16]: `owner` redundant here

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1,25 +1,84 @@
...
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: SlideLimitationValidator, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+    ) -> None:
```
Comment (Vasyl-Smutok) [inline app/main.py:16]:
```suggestion
    def __get__(self, instance: SlideLimitationValidator) -> None:
```

File: app/main.py [hunk 3]
```diff
@@ -1,25 +1,84 @@
...
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: SlideLimitationValidator,
+            value: int
+    ) -> None:
```
Comment (Vasyl-Smutok) [inline app/main.py:23]:
```suggestion
    def __set__(self, instance: SlideLimitationValidator, value: int) -> None:
```

File: app/main.py [hunk 4]
```diff
@@ -1,25 +1,84 @@
...
+        if not isinstance(value, int):
+            raise TypeError(
+                "Value should be integer"
+            )
```
Comment (Vasyl-Smutok) [inline app/main.py:27]:
```suggestion
            raise TypeError("Value should be integer")
```

**Review by vsmutok — APPROVED** [review]
Good Job 👍 
also check comments

**Pull request closed without merging**